osgar/logger.py

```diff
@@ -147,13 +165,23 @@ def write(self, stream_id, data):
     parser = argparse.ArgumentParser(description='Extract data from log')
     parser.add_argument('logfile', help='filename of stored file')
     parser.add_argument('--stream', help='stream ID', type=int, default=None)
+    parser.add_argument('--name', help='stream name', default=None)
+    parser.add_argument('--list-names', '-l', help='list stream names', action='store_true')
     parser.add_argument('--times', help='display timestamps', action='store_true')
     parser.add_argument('--raw', help='skip data deserialization',
                         action='store_true')
     args = parser.parse_args()
 
+    only_stream = args.stream
+    if args.name is not None:
+        only_stream = lookup_stream_id(args.logfile, args.name)
+
+    if args.list_names:
+        print(lookup_stream_names(args.logfile))
+        sys.exit()
+
     with LogReader(args.logfile) as log:
-        for timestamp, stream_id, data in log.read_gen(args.stream):
+        for timestamp, stream_id, data in log.read_gen(only_stream):
```

Comment:
So the option `--stream` is no longer supported? If it is not supported, it should be removed.

Reply:
Both options are still supported `--stream <number>` as well as `--name <named stream>`, but you are right, that this leads to unnecessary rich interface. What about re-use `--stream <name>` where hash (#) would be used for old integer indexing, i.e. `--stream #0` would be original "service stream"? [somehow I would still like to keep indexing via numbers, similarly to ADTF pins]

Reply:
Ok, I missed line 175. I would just checked if it is number or not and act based on that. That would mean you cannot have a stream named "0" but that is actually a good thing, isn't it?

Reply:
OK, so it will be backward compatible + allow also names as input. Agree.

Reply:
👍